homeassistant/components/shelly/sensor.py

```diff
@@ -2,15 +2,31 @@
 import aioshelly
 
 from homeassistant.components import sensor
-from homeassistant.const import TEMP_CELSIUS, TEMP_FAHRENHEIT, UNIT_PERCENTAGE
+from homeassistant.const import (
+    ELECTRICAL_CURRENT_AMPERE,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+    UNIT_PERCENTAGE,
+    VOLT,
+)
 from homeassistant.helpers.entity import Entity
 
 from . import ShellyBlockEntity, ShellyDeviceWrapper
 from .const import DOMAIN
 
 SENSORS = {
+    "battery": [UNIT_PERCENTAGE, sensor.DEVICE_CLASS_BATTERY],
+    "current": [ELECTRICAL_CURRENT_AMPERE, sensor.DEVICE_CLASS_CURRENT],
+    "deviceTemp": [None, sensor.DEVICE_CLASS_TEMPERATURE],
+    "energy": [ENERGY_KILO_WATT_HOUR, sensor.DEVICE_CLASS_ENERGY],
+    "energyReturned": [ENERGY_KILO_WATT_HOUR, sensor.DEVICE_CLASS_ENERGY],
     "extTemp": [None, sensor.DEVICE_CLASS_TEMPERATURE],
     "humidity": [UNIT_PERCENTAGE, sensor.DEVICE_CLASS_HUMIDITY],
+    "overpowerValue": [POWER_WATT, sensor.DEVICE_CLASS_ENERGY],
```

Comment:
I don't think this is device class energy? It's overpower and so just device class `None` ?

Reply:
I meant DEVICE_CLASS_POWER 🤦 Overpower is in watts.